### ISSUE TITLE
Are empty weapon names valid?

### ISSUE THREAD

**Issue by niffiwan** [scp-fs2open/fs2open.github.com#400]
While testing something, I found that vassago-01 now triggers an assert on load. The assert was introduced by 3b43614ae, and it's because a ship has had some weapons removed from its turrets.

```
$Name: Melchom		;! Object #10
...
+Subsystem: turret08
+Secondary Banks: ( "" )
+Sbank Ammo: ( 0 )
+Subsystem: turret09
+Secondary Banks: ( "" )
+Sbank Ammo: ( 0 )
```

The question is; is this a valid way of removing weapons? Before turret ammo was introduced I'd guess it was the only way of removing weapons. 

Should the mission be changed?
Should the code be changed to skip turrets that don't have any weapons?

**Comment by niffiwan** [comment]
Maybe I've answered my own question.  -1 is a valid value for a weapon, denoting that the bank is empty. The Assert itself is potentially doing a negative array read into Weapon_info...

**Comment by Goober5000** [comment]
This is valid mission text.  Setting weapon names to blank is the usual way to remove weapons from a ship, and it has a long precedent.  Look at Vasudan Freighter 2 in the FS1 (or FSPort) mission sm1-03. ;)

**Comment by niffiwan** [comment]
cool; I've created a fix in #401
